internal/clients/team3/client.go

```diff
@@ -17,7 +18,7 @@ type client struct {
 }
 
 func (c *client) DemoEvaluation() {
-	evalResult, err := common.BasicRuleEvaluator("Kinda Complicated Rule")
+	evalResult, err := rules.BasicBooleanRuleEvaluator("Kinda Complicated Rule")
```

Comment:
Have we decided on how to enumerate different rules with constants/enuns

Reply:
We still run into the same problem as before in terms of extendability to runtime rule generation, Like I'm unsure if we can add to an enum at runtime?

Reply:
Hmm, true! But we need to track all the names, right? What if someone suggests a new rule with the same name?

Reply:
what about rule variables btw?

Reply:
For the same name issue, we refuse to register and return an error that the clients can deal with I guess. But I do agree it would be nice to address them with something other than strings. I'm unsure what we can use though.

Reply:
Variables can be registered before or at runtime and also have the same error returning function to prevent multiple registers. They can be modified with a specific function

Reply:
As long as the island can "look at" the variables at runtime I think it should be fine